### DISCUSSION TITLE
[External project "assimp"] Status of models in usd-wg/assets

### DISCUSSION THREAD

**Discussion by tellypresence** [syoyo/tinyusdz#178] [Show and tell]
Recently support for models from [`usd-wg/assets/full_assets`](https://github.com/usd-wg/assets/tree/main/full_assets) has improved; previously none of the assets would load, but in recent `dev` branch most now load to some extent.

<table>
<tr><th>Folder</th><th>Model</th><th>Screenshots</th><th>Notes</th></tr>
<tr><td><tt>CarbonFrameBike</tt></td><td><a href="https://github.com/usd-wg/assets/blob/main/full_assets/CarbonFrameBike/CarbonFrameBike.usdz"><tt>CarbonFrameBike.usdz</tt></a></td><td><img src="https://github.com/user-attachments/assets/71dd39b0-14bc-405a-8514-0d62aa2c530c" alt="usd_wg_assets_CarbonFrameBike" width=180 /></td><td>Exploded geometry</td></tr>
<tr><td><tt>ElephantWithMonochord</tt></td><td><a href="https://github.com/usd-wg/assets/blob/main/full_assets/ElephantWithMonochord/SoC-ElephantWithMonochord.usdc"><tt>SoC-ElephantWithMonochord.usdc</tt></a></td><td><img src="https://github.com/user-attachments/assets/72ba1b2d-0b97-48fc-abaa-4286d79e3e38" alt="usd_wg_assets_ElephantWithMonochord" width=180 /></td><td>UVs incorrect, objects seem to be picking up a single pixel from textures for color</td></tr>

<tr><td rowspan=2><tt>OpenChessSet</tt></td><td><a href="https://github.com/usd-wg/assets/blob/main/full_assets/OpenChessSet/assets/King/King_geom.usd"><tt>assets/King/King_geom.usd</tt></a></td><td><img src="https://github.com/user-attachments/assets/73f331ee-0fee-4e28-8fc5-48add58986f6" alt="usd_wg_assets_OpenChessSet_assets_King" width=180 /></td><td rowspan=2>Not picking up the main scene, only sub-objects; none of the assets seem to have normals</td></tr>
<tr><td><a href="https://github.com/usd-wg/assets/blob/main/full_assets/OpenChessSet/assets/Knight/Knight_geom.usd"><tt>assets/Knight/Knight_geom.usd</tt></a></td><td><img src="https://github.com/user-attachments/assets/3a5ca78c-e91d-4a28-903b-328f7c6c8dde" alt="usd_wg_assets_OpenChessSet_assets_Knight" width=180 /></td></tr>

<tr><td rowspan=2><tt>StandardShaderBall</tt></td><td><a href="https://github.com/usd-wg/assets/blob/main/full_assets/StandardShaderBall/material_surface_geo/subdiv.usda"><tt>material_surface_geo/subdiv.usda</tt></a></td><td><img src="https://github.com/user-attachments/assets/4e8a50ad-9add-4d5a-a7f3-88dbb0c1b875" alt="usd_wg_assets_StandardShaderBall_material_surface_geo_subdiv" width=180 /></td><td rowspan=2>Not picking up the main scene, only sub-objects; these objects are "inside out" by default -- flipping winding order turns them right-side-out, but causes normals to be 180 degrees off</td></tr>
<tr><td><a href="https://github.com/usd-wg/assets/blob/main/full_assets/StandardShaderBall/material_surface_geo/triangulated.usda"><tt>material_surface_geo/triangulated.usda</tt></a></td><td><img src="https://github.com/user-attachments/assets/8d6679b9-ad8b-4851-b8e2-0eeb1eda3e73" alt="usd_wg_assets_StandardShaderBall_material_surface_geo_triangulated" width=180 /></td></tr>
<tr><td rowspan=2><tt>Teapot</tt></td><td><a href="https://github.com/usd-wg/assets/blob/main/full_assets/Teapot/geo/FancyTeapot.usd"><tt>geo/FancyTeapot.usd</tt></a></td><td><img src="https://github.com/user-attachments/assets/ff63d5a4-6496-4a49-a410-d45dd4442047" alt="usd_wg_assets_Teapot_geo_FancyTeapot" width=180 /></td><td rowspan=2>Not picking up the main scene, only sub-objects; note that FancyTeapot has no normals</td></tr>
<tr><td><a href="https://github.com/usd-wg/assets/blob/main/full_assets/Teapot/geo/UtahTeapot.usd"><tt>geo/UtahTeapot.usd</tt></a></td><td><img src="https://github.com/user-attachments/assets/0bdaaa00-a4d0-40e8-8e98-f1f8a1d11881" alt="usd_wg_assets_Teapot_geo_UtahTeapot" width=180 /></td></tr>
<tr><td><tt>UsdCookie</tt></td><td><a href="https://github.com/usd-wg/assets/blob/main/full_assets/UsdCookie/UsdCookie.usdz"><tt>UsdCookie.usdz</tt></a></td><td><img src="https://github.com/user-attachments/assets/bb31e43b-7ea8-4878-b06a-aa02d9cd55ef" alt="UsdCookie" width=180 /></td><td>(Note: 8k texture provides good detail)</td></tr>
</table>

**Comment by syoyo** [comment]
You should not post a discussion about external project.
